scripts/azure_cleanup.py

```
@@ -0,0 +1,69 @@
+    with open(output, 'w') as report:
+        report.write('azure_cleanup.py, NICs and PIPs Cleanup')
+        report.write("\nDate: {}\n".format(datetime.now()))
+        try:
+            for provider_key in list_provider_keys('azure'):
+                provider_mgmt = get_mgmt(provider_key)
+                nic_list = provider_mgmt.list_free_nics(nic_template)
+                pip_list = provider_mgmt.list_free_pip(pip_template)
+                stack_list = provider_mgmt.list_stack(days_old=days_old)
+                report.write("----- Provider: {} -----".format(provider_key))
+                if nic_list:
+                    report.write("\nRemoving Nics with the name \"{}\":\n".format(nic_template))
+                    report.write("\n".join(str(k) for k in nic_list))
+                    provider_mgmt.remove_nics_by_search(nic_template)
+                else:
+                    report.write("\nNo \"{}\" NICs were found\n".format(nic_template))
+                if pip_list:
+                    report.write("\nRemoving Public IPs with the name \"{}\":\n".
+                                 format(pip_template))
+                    report.write("\n".join(str(k) for k in pip_list))
+                    provider_mgmt.remove_pips_by_search(pip_template)
+                else:
+                    report.write("\nNo \"{}\" Public IPs were found\n".format(pip_template))
+                if stack_list:
+                    report.write(
+                        "\nRemoving empty Stacks:")
+                    for stack in stack_list:
+                        if provider_mgmt.is_stack_empty(stack):
+                            provider_mgmt.delete_stack(stack)
+                            report.write("\nStack {} is empty - Removed".format(stack))
+                else:
+                    report.write("\nNo stacks older than \"{}\" days were found\n".format(
+                        days_old))
+            return 0
+        except Exception:
```

REQUIRED: if you decided to catch all exceptions, I guess you have to put exception data not only to stdout but to report file as well. Otherwise, I don't see sense in catching expections at all.